### PR TITLE
Reorder effects

### LINKED ISSUE #3749: Allow re-ordering of effects (object/layers)
## Description

I recently learned that the way effects are ordered will change how the effects are applied.  

For example, "glow + outline" looks different than "outline + glow".

If users want to adjust the order of effects today, they need to delete and recreate the effects. 

## Solution suggested

Allow users to change the order that effects (both objects and layers) are listed.  

Preferred method: Drag and drop the effects to reorder

Alternative method: Add "Move Up" and "Move Down" in a right-click menu

### PR DESCRIPTION
**Pull request by AlexandreSi**
Fixes #3749

Add checks on names as well

demo_effects_list

### CODE REVIEW

File: newIDE/app/src/EffectsList/index.js [hunk 1]
```diff
@@ -144,118 +204,197 @@ export default function EffectsList(props: Props) {
...
-                                  <MarkdownText
-                                    source={effectMetadata.description}
+                          {({
+                            connectDragSource,
+                            connectDropTarget,
+                            isOver,
+                            canDrop,
+                          }) =>
+                            connectDropTarget(
+                              <div
```
Comment (ClementPasteau) [inline newIDE/app/src/EffectsList/index.js:239]: should those divs be <Columns> and <Lines> ?

File: newIDE/app/src/EffectsList/index.js [hunk 2]
```diff
@@ -144,118 +204,197 @@ export default function EffectsList(props: Props) {
...
+                                key={effect.ptr}
+                                style={{
```
Comment (ClementPasteau) [inline newIDE/app/src/EffectsList/index.js:241]: seeing inline style makes me wonder if we're doing something wrong

**Review by ClementPasteau — APPROVED** [review]
lgtm overall! quick question in comments

File: newIDE/app/src/EffectsList/index.js (hunk 2 above)
Comment (AlexandreSi) [reply newIDE/app/src/EffectsList/index.js:241]: You cannot give a React component to the connect functions of the drag n drop, they require pure HTML component :/

**Comment by 4ian** [comment]
LGTM!

File: newIDE/app/src/EffectsList/index.js (hunk 2 above)
Comment (ClementPasteau) [reply newIDE/app/src/EffectsList/index.js]: 👍

**Pull request merged**